File: Task12/Animal.m
//
//  Animal.m
//  Task12
//
//  Created by Sergey Plotnikov on 07.07.16.
//  Copyright © 2016 Sergey Plotnikov. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "Animal.h"

@interface Animal ()

@property(nonatomic) NSInteger calories;
@property(nonatomic) NSMutableArray *mStomach;

@end

@implementation Animal

@synthesize name = _name;
@synthesize stomach = _stomach;
@synthesize calories = _calories;

- (NSArray *)stomach
{
    return [self.mStomach copy];
}

- (instancetype)init
{
    self = [super init];
    
    if (self) {
        _mStomach = [NSMutableArray array];
    }
    return self;
}

- (void)eat:(Creature *)creature withCalories:(NSInteger)calories
{
    self.calories += calories;
    [self.mStomach addObject: creature];
}

- (BOOL)isProtected
{
    return arc4random()%2 == 0;
}

- (NSString *)description
{
    return [NSString stringWithFormat:@"%@(%ld)", self.name, self.calories];
}

@end

File: Task1_2/Animal.swift
//
//  Animal.swift
//  Task1_2
//
//  Created by Admin on 22/09/16.
//  Copyright © 2016 Admin. All rights reserved.
//

import Foundation

class Animal: Creature, Nameable {
    
    private (set) var name: String
    private (set) var stomack: [Creature]
    
   
    init (name: String, calories: Int) {
        self.name = name
        stomack = [Creature]()
        super.init(calories: calories)
    }
    
    func isProtected() -> Bool {
        return arc4random_uniform(2) == 0
    }
    
    func eat(creature: Creature, calories: Int) {
        self.calories += calories
        stomack.append(creature)
    }
    
    func showStomack(spacePrefix: String = "") -> String {
        var s = spacePrefix + name + ":\n"
        let nextSpacePrefix = spacePrefix + "  "
        for creature in stomack {
            let creatureDescription: String = creature is Animal ? (creature as! Animal).showStomack(spacePrefix: nextSpacePrefix) : nextSpacePrefix + creature.description
            s += creatureDescription
            if !s.hasSuffix("\n") {
                s += "\n"
            }
            
        }
        return s
    }
    
}
